core/src/main/java/feast/core/validators/FeatureTableValidator.java

```diff
@@ -0,0 +1,66 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ * Copyright 2018-2019 The Feast Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package feast.core.validators;
+
+import static feast.core.validators.Matchers.*;
+
+import feast.proto.core.FeatureProto.FeatureSpecV2;
+import feast.proto.core.FeatureTableProto.FeatureTableSpec;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+public class FeatureTableValidator {
+  protected static final Set<String> RESERVED_NAMES =
+      Set.of("created_timestamp", "event_timestamp");
+
+  public static void validateSpec(FeatureTableSpec spec) {
```

Comment:
does it check for empty features/entities list?

Reply:
Added check in same file.